computers/computers/spiders/computer_spider.py

```diff
@@ -0,0 +1,86 @@
+import scrapy
+from computers.item_loaders import ComputerItemLoader
+from computers.items import ComputerItem
+
+
+class ComputerSpider(scrapy.Spider):
+    name = "computer_spider"
+    STARTING_PAGE_NUMBER = 1
+    start_urls = [
+        f"https://www.komputronik.pl/search-filter/5801/komputery-do-gier?p={STARTING_PAGE_NUMBER}"
+    ]
+    custom_settings = {"AUTOTHROTTLE_ENABLED": True, "DOWNLOAD_DELAY": 6}
+    PROCESSOR = "Procesor"
+    GRAPHICS = "Karta graficzna"
+    CHIPSET = "Płyta główna"
+    PART_TUPLE = (PROCESSOR, GRAPHICS, CHIPSET)
+
+    def __get_processor(self, response):
+        for row in response.xpath("//tr"):
+            if row.xpath(".//th/text()").get() == "Seria procesora":
+                return row.xpath(".//td/text()").get()
+
+        return
+
+    def __get_graphics(self, response):
+        for row in response.xpath("//tr"):
+            if row.xpath(".//th/text()").get() == self.GRAPHICS:
+                return row.xpath(".//td/a/text()").get(row.xpath(".//td/text()").get())
+
+        return
+
+    def __get_chipset(self, response):
+        for row in response.xpath("//tr"):
+            if row.xpath(".//th/text()").get() == "Chipset płyty głównej":
+                return row.xpath(".//td/text()").get()
+
+        return
+
+    def parse(self, response, **kwargs):
+        for element in response.xpath('.//div[@class="pe2-head"]'):
+            url = element.xpath(".//a/@href").get()
+            if url:
```

Comment:
[element for element in response.xpath('.//div[@class="pe2-head"]//a/@href' if element)]

Reply:
response.follow_all()